mac os x support

fretn opened the pull request:
This PR adds support for Mac OS X, its tested on OS X 10.9 and 10.11

The only thing that currently doesn't work is ctrl-c to close the program

adrienverge commented:
Thanks for sharing this @fretn!

Could you cleanup and squash the commits together?

If you have trouble with your text editor and indentation, you can use the `astyle` command to automatically format your code, [as shown here](https://github.com/adrienverge/openfortivpn/blob/3e08fdc/tests/lint/astyle.sh#L15) ;-)

adrienverge commented on src/io.c:
```diff
@@ -161,7 +170,11 @@ static void *pppd_read(void *arg)
 			sem_post(&sem_pppd_ready);
+#else
+			dispatch_semaphore_signal(sem_pppd_ready);
+#endif
```
Could we avoid all these `#ifndef __APPLE__` by defining a macro, like:

```c
#ifdef __APPLE__
#undef sem_post
#define sem_post(s) semaphore_signal(*s)
...
```
?

PS: Does defining the `USE_POSIX_SEM` constant provide `sem_post()`?

fretn commented:
I'll make a new PR where we have one commit, and where I remove some #ifdefs and replaced it with your example

about USE_POSIX_SEM I need to look that up
Unnamed semaphores don't work on osx, so I switched to grand central dispatch to use semaphores. Migrating the code from sem_init (unnamed) to sem_open (named), might solve this problem too

about the indents, sorry about that, my vim settings are messed up, in case you use vim, what are your tab settings ?

adrienverge commented:
Please do not make a new PR, just squash your commits and `push --force`.

> about USE_POSIX_SEM I need to look that up

Sure, please do! The less diff, the better.

> about the indents, sorry about that, my vim settings are messed up, in case you use vim, what are your tab settings ?

I do this kind of manually (with `au FileType c setlocal noexpandtab` and
completing with spaces) but I think there are Vim scripts to respect the Linux kernel coding style.

PS: I just tested [vivien/vim-linux-coding-style](https://github.com/vivien/vim-linux-coding-style), it seems to work great!

fretn commented:
sorry, didn't see your message, I already started over because I messed up my fork (I only know the basic git stuff)

so how do I proceed now ? you want me to show you a diff before I start a merge request ?

fretn commented:
https://github.com/fretn/openfortivpn/commit/72bb9c75483d3a697e35cd46231b796333f6e09c

adrienverge commented:
Let's keep all this in the same pull request, as it eases tracking changes and comments. To overwrite this specific pull request, you'll need to [*squash*/*fixup* your commits into one](https://www.google.fr/webhp?&ion=1&espv=2#q=git%20squash) and `git push -f` to the branch you opened the PR from.

fretn commented:
thats the problem, I deleted my fork because the squash didn't work out. I didn't know i had to force push, so I started messing around :-1:
